src/main/java/frc/robot/subsystems/ball/feeder/UltrasonicPoller.java

```diff
@@ -9,22 +10,50 @@
 class UltrasonicPoller extends RepeatingPooledSubsystem {
 
     Ultrasonic ultrasonicSensor;
+    StopWatch timer = new StopWatch();
 
-    UltrasonicPoller(Ultrasonic sensor, int time) {
+    // Consider renaming, couldn't think of a better one ATM
+    boolean noBalls;
+    long time;
+    double distance;
+    double currentRange;
+
+    UltrasonicPoller(Ultrasonic sensor, long time, double distance, long rate) {
         // sensor is pretty self explanatory
         // time is how long you want the sensor to not detect an object until true is returned
-        super(25, TimeUnit.MILLISECONDS); // 4Hz seems reasonable
+        // distance is the minimum distance threshold
+        super(rate, TimeUnit.MILLISECONDS); // 4Hz seems reasonable
         this.ultrasonicSensor = sensor;
+        this.time = time;
+        this.distance = distance;
+        this.noBalls = false;
+
+    }
+
+    UltrasonicPoller(Ultrasonic sensor, long time, double distance) {
+        this(sensor, time, distance, 25);
```

Comment:
What's the reason for having both of these constructors?

Reply:
Default refresh rate

Reply:
I can see that, but why do you need to have a constructor for a default rate and another that allows it to be specified by whoever's constructing it?

Reply:
True. Considering this was potentially going to be merged into Yggdrasil, I thought that I might as well do it semi-right the first time.

Reply:
Just keep it clean and simple for now. There's going to be other changes needed before upstreaming anyway, so don't try to plan too far in advance.

Reply:
Ok, I'll just leave it there though, no point in removing it if it's already there.